plugin/evm/vm.go

```diff
@@ -513,8 +514,13 @@ func (vm *VM) GetBlock(id ids.ID) (snowman.Block, error) {
 
 // SetPreference sets what the current tail of the chain is
 func (vm *VM) SetPreference(blkID ids.ID) {
-	err := vm.chain.SetTail(common.Hash(blkID))
-	vm.ctx.Log.AssertNoError(err)
+	block := vm.getBlock(blkID)
+	if block == nil {
+		vm.ctx.Log.Error("could not get preference block %s", blkID)
```

Comment:
Could you change this to `vm.ctx.Log.AssertNoError(err)`? I think we should treat this as a critical error